### ISSUE TITLE
startMO2: Command not found

### ISSUE THREAD

**Issue by hftsai256** [sonic2kk/steamtinkerlaunch#642]
## System Information
- SteamTinkerLaunch version: v11.12.20221024-5
- Distribution: Steam Deck: SteamOS
- Installation Method: Installed manually from source downloaded in "release" section as instructed in [this wiki page](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Installation#steam-deck). Not through ProtonUp-Qt. A symbolic link under `$PATH` was created.

## Issue Description
Could not start mo2 from menu, and will get the following error when it is launched from terminal.

```
(deck@steamdeck ~)$ steamtinkerlaunch mo2 s
Preparing to install SteamTinkerLaunch on Steam Deck
Updating SteamTinkerLaunch to latest git

Dependency 'wget' already installed, nothing to do.
Dependency 'innoextract' already installed, nothing to do.
Dependency 'cabextract' already installed, nothing to do.
Dependency 'yad' already installed, nothing to do.

Finished updating SteamTinkerLaunch ('v11.12.20221024-5')!
Wed Oct 26 10:21:28 PM EDT 2022 INFO - startMO2 - Starting '/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe' in standalone mode with command:
Wed Oct 26 10:21:28 PM EDT 2022 INFO - startMO2 - WINEDEBUG="-all" WINEPREFIX="/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx" "" "/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe"
/home/deck/.local/bin/steamtinkerlaunch: line 13450: : command not found
```

## Logs
```
Wed Oct 26 10:31:32 PM EDT 2022 INFO - setflatpak - started SteamTinkerLaunch from /home/deck/.local/bin/steamtinkerlaunch
Wed Oct 26 10:31:32 PM EDT 2022 INFO - initAID - Set AID to PLACEHOLDERAID '31337'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLangFile - Language from command line is 'english'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLangFile - SYSTEMSTLCFGDIR is '/usr/share/steamtinkerlaunch'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLangFile - Command line language 'english' is no file - trying to find its absolute path
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLangFile - Loading language file from script directory '/home/deck/stl/prefix/lang/english.txt'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLanguage - Loading STLLANG from '/home/deck/.config/steamtinkerlaunch/global.conf'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - loadLanguage - STLLANG from '/home/deck/.config/steamtinkerlaunch/global.conf' is 'english'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - steamdedeckt - Seems like we have a Steam Deck here - making some specific settings
Wed Oct 26 10:31:33 PM EDT 2022 INFO - setSteamPaths - Reading Steam Path variables from '/dev/shm/steamtinkerlaunch/steampaths.txt'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - steamdedeckt - Did not detect a running 'gamescope' process - assuming we're running in Desktop Mode
Wed Oct 26 10:31:33 PM EDT 2022 INFO - steamdedeckt - Set 'FIXGAMESCOPE' to '0'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - steamdedeckt - Set 'SMALLDESK' to '1'
Wed Oct 26 10:31:33 PM EDT 2022 INFO - gitUpdate - Pulling 'https://github.com/sonic2kk/steamtinkerlaunch' update in '/home/deck/stl/prefix'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - installFilesSteamDeck - Fetching from git seems to have succeeded ok.
Wed Oct 26 10:31:34 PM EDT 2022 INFO - installDependencyVersionFromURL - Using 'wget' binary found in path: '/usr/bin/wget'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - installDependencyVersionFromURL - Using 'innoextract' binary found in path: '/home/deck/stl/deps/usr/bin/innoextract'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - installDependencyVersionFromURL - Using 'cabextract' binary found in path: '/home/deck/stl/deps/usr/bin/cabextract'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - checkSteamDeckDependencies - Using yad binary found in path: '/home/deck/stl/deps/usr/bin/yad'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - CompatTool - '/home/deck/.local/share/Steam/compatibilitytools.d/SteamTinkerLaunch' already exists - checking if 'steamtinkerlaunch' symlink needs to be updated
Wed Oct 26 10:31:34 PM EDT 2022 SKIP - CompatTool - Nothing to do the '/home/deck/.local/share/Steam/compatibilitytools.d/SteamTinkerLaunch/steamtinkerlaunch' symlink still points to '/home/deck/stl/prefix/steamtinkerlaunch'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - steamdedeckt - backup '/home/deck/.config/gtk-3.0/gtk.css' to '/home/deck/.config/gtk-3.0/gtk.css_ORIGNAL'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - steamdedeckt - adding bigger scrollbar and customising some other UI elements using '/home/deck/.config/gtk-3.0/gtk.css'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - getCurrentCommandline - No Proton in command line arguments 'mo2 s'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - saveOrgVars - Storing some original variables to restore them later
Wed Oct 26 10:31:34 PM EDT 2022 INFO - emptyVars - Emptied some original variables as they slowdown several system calls when started from steam
Wed Oct 26 10:31:34 PM EDT 2022 INFO - emptyVars - Set $PATH to '/usr/local/sbin:/usr/local/bin:/usr/bin:/var/lib/flatpak/exports/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/deck/.local/bin:/home/deck/stl/deps/usr/bin:'
Wed Oct 26 10:31:34 PM EDT 2022 START - ######### Initializing Game Launch 31337 using SteamTinkerLaunch v11.12.20221024-5 #########
Wed Oct 26 10:31:34 PM EDT 2022 INFO - main - Starting with a clean log
Wed Oct 26 10:31:34 PM EDT 2022 INFO - main - Start creating default configs
Wed Oct 26 10:31:34 PM EDT 2022 INFO - createDefaultCfgs - START
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLangFile - Language from command line is 'english'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLangFile - GLOBALSTLLANGDIR is '/home/deck/stl/prefix/lang'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLangFile - Command line language 'english' is no file - trying to find its abolute path
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLangFile - Loading language file from script directory '/home/deck/stl/prefix/lang/english.txt'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLanguage - Loading STLLANG from '/home/deck/.config/steamtinkerlaunch/global.conf'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - loadLanguage - STLLANG from '/home/deck/.config/steamtinkerlaunch/global.conf' is 'english'
Wed Oct 26 10:31:34 PM EDT 2022 INFO - createDefaultCfgs - setSteamPaths:
Wed Oct 26 10:31:34 PM EDT 2022 INFO - setSteamPaths - Reading Steam Path variables from '/dev/shm/steamtinkerlaunch/steampaths.txt'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - getGameOS - Starting game OS detection
Wed Oct 26 10:31:35 PM EDT 2022 INFO - getGameOS - STEAM_COMPAT_DATA_PATH is not defined, so this is either a Linux Game or no game was started at all
Wed Oct 26 10:31:35 PM EDT 2022 INFO - listAllSettingsEntries - '/dev/shm/steamtinkerlaunch/setentries.txt' already exists - nothing to do
Wed Oct 26 10:31:35 PM EDT 2022 INFO - createDefaultCfgs - STOP
Wed Oct 26 10:31:35 PM EDT 2022 SKIP - listSteamLibraries - not recreating already available '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setSteamLibraryPaths - STEAM_COMPAT_LIBRARY_PATHS set to '/home/deck/.local/share/Steam/steamapps:/run/media/mmcblk0p1/steamapps'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setSteamLibraryPaths - STEAM_COMPAT_MOUNTS set to '/home/deck/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier:/home/deck/.local/share/Steam/steamapps/common/Steamworks Shared'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setSteamLibraryPaths - STEAM_COMPAT_TOOL_PATHS set to '/home/deck/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier:/home/deck/.local/share/Steam/steamapps/common/Steamworks Shared'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - main - Checking internal dependencies:
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setAwkBin - Found 'gawk' as an 'awk' variant. It should work without any issues, because 'gawk' was tested completely
Wed Oct 26 10:31:35 PM EDT 2022 INFO - checkIntDeps - Skipping yad version check on SteamDeck
Wed Oct 26 10:31:35 PM EDT 2022 INFO - main - Initializing first Proton:
Wed Oct 26 10:31:35 PM EDT 2022 INFO - initOldProtonArr - Creating an initial array with available Proton versions using the file '/dev/shm/steamtinkerlaunch/ProtonCSV.txt' which was created during a previous run
Wed Oct 26 10:31:35 PM EDT 2022 INFO - initFirstProton - Initializing Proton
Wed Oct 26 10:31:35 PM EDT 2022 INFO - initFirstProton - Initial Proton version 'FIRSTUSEPROTON' from '/home/deck/.config/steamtinkerlaunch/default_template.conf' is 'proton-7.0-4'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - main - Initializing default window resolution
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setInitWinXY - Using '1280x800' from config '/dev/shm/steamtinkerlaunch/defres.txt'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - main - Checking command line: incoming arguments 'mo2 s'
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setMO2Vars - Using GE-Proton7-38 for ModOrganizer
Wed Oct 26 10:31:35 PM EDT 2022 ERROR - setModWine - MO2WINE was not found - can't continue
Wed Oct 26 10:31:35 PM EDT 2022 INFO - prepareMO2 - Updating/creating ModOrganizer instances
Wed Oct 26 10:31:35 PM EDT 2022 INFO - setMO2Vars - Using GE-Proton7-38 for ModOrganizer
Wed Oct 26 10:31:35 PM EDT 2022 ERROR - setModWine - MO2WINE was not found - can't continue
Wed Oct 26 10:31:35 PM EDT 2022 SKIP - listSteamLibraries - not recreating already available '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Wed Oct 26 10:31:35 PM EDT 2022 SKIP - setModGameSyms - '/run/media/mmcblk0p1/steamapps/compatdata/377160/pfx' missing - looks like the game was removed or never started yet
Wed Oct 26 10:31:36 PM EDT 2022 INFO - setMO2Vars - Using GE-Proton7-38 for ModOrganizer
Wed Oct 26 10:31:36 PM EDT 2022 ERROR - setModWine - MO2WINE was not found - can't continue
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - manageMO2GInstance - argument 1 '' is invalid
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - listSteamLibraries - not recreating already available '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - setModGameSyms - '/run/media/mmcblk0p1/steamapps/compatdata/377160/pfx' missing - looks like the game was removed or never started yet
Wed Oct 26 10:31:36 PM EDT 2022 INFO - prepareMO2 - No log '/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/users/steamuser/AppData/Local/ModOrganizer/nxmhandler.log' found, so the last used instance can't be determined - using last supported instance installed instead
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - setModGameSyms - '/run/media/mmcblk0p1/steamapps/compatdata/377160/pfx' missing - looks like the game was removed or never started yet
Wed Oct 26 10:31:36 PM EDT 2022 INFO - setMO2Vars - Using GE-Proton7-38 for ModOrganizer
Wed Oct 26 10:31:36 PM EDT 2022 ERROR - setModWine - MO2WINE was not found - can't continue
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - manageMO2GInstance - argument 1 '' is invalid
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - listSteamLibraries - not recreating already available '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - setModGameSyms - '/run/media/mmcblk0p1/steamapps/compatdata/377160/pfx' missing - looks like the game was removed or never started yet
Wed Oct 26 10:31:36 PM EDT 2022 INFO - prepareMO2 - No log '/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/users/steamuser/AppData/Local/ModOrganizer/nxmhandler.log' found, so the last used instance can't be determined - using last supported instance installed instead
Wed Oct 26 10:31:36 PM EDT 2022 SKIP - setModGameSyms - '/run/media/mmcblk0p1/steamapps/compatdata/377160/pfx' missing - looks like the game was removed or never started yet
Wed Oct 26 10:31:36 PM EDT 2022 INFO - prepareMO2 - No ModOrganizer instance provided
Wed Oct 26 10:31:36 PM EDT 2022 INFO - setMO2Vars - Using GE-Proton7-38 for ModOrganizer
Wed Oct 26 10:31:36 PM EDT 2022 ERROR - setModWine - MO2WINE was not found - can't continue
Wed Oct 26 10:31:36 PM EDT 2022 INFO - setMO2DLMime - Desktopfile '/home/deck/.local/share/applications/ModOrganizer-steamtinkerlaunch-dl.desktop' looks to be up2date
Wed Oct 26 10:31:36 PM EDT 2022 INFO - startMO2 - Starting '/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe' in standalone mode with command:
Wed Oct 26 10:31:36 PM EDT 2022 INFO - startMO2 - WINEDEBUG="-all" WINEPREFIX="/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx" "" "/home/deck/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe"
Wed Oct 26 10:31:36 PM EDT 2022 INFO - restoreGtkCss - recovering original gtk.css from '/home/deck/.config/gtk-3.0/gtk.css_ORIGNAL'
```

**Comment by frostworx** [comment]
seems like you chose GE-Proton7-38 as "mod organizer 2 proton", but the wine binary shipped with GE-Proton7-38 could not be found for some reason _(not reproducable here)_

try one or a combination of:
- check in `/dev/shm/steamtinkerlaunch/ProtonCSV.txt` if the proton path for GE-Proton7-38 is valid
- remove `/dev/shm/steamtinkerlaunch/ProtonCSV.txt` and let it recreate automatically _(and do the first check again)_
- define a new "mod organizer 2 proton" in the global menu

before a new `steamtinkerlaunch mo2 s` attempt

**Comment by sonic2kk** [comment]
Hi @hftsai256, Frostworx has given some good advice here. Just to add to this, if you still have problems after those steps, try deleting your `/dev/shm/steamtinkerlaunch` folder, and if you still have problems then just double check the MO2 Proton version defined in the global menu, pictured below.

![image](https://user-images.githubusercontent.com/7917345/198356126-c6bea8ae-98de-4fc2-9258-4d1a4f4c77f4.png)

It is also possible, though unlikely, that you installation of GE-Proton is invalid (i.e., the "proton" script doesn't exist). Does GE-Proton7-38 work for you in other games? If so that would almost certainly rule out an invalid installation, but just worth checking :-)

**Comment by hftsai256** [comment]
Thanks for all the suggestions and instructions to look into. I first looked into `/dev/shm/steamtinkerlaunch` and found the path recorded in `ProtonCSV.txt` are pointing to proper locations. Also I could launch other games with GE-Proton7-38 so the installation for proton environment itself seems fine.

Its compatdata location is also correct in global menu. I ended up blowing out the entire `~/.config/steamtinkerlaunch` environment and MO2 can now launched properly through command line after the environment got regenerated and reinstalled.

Actual operating steps:
```
rm -rf ~/.config/steamtinkerlaunch
steamtinkerlaunch mo2 s
```
